docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+    migplan: test
+  name: mtest
+  namespace: openshift-migration
+  labels:
+    migplan: test
+spec:
+  analyzeImageCount: true
+  analyzeK8SResources: true
+  analyzePVCapacity: true
+  migPlanRef:
+    name: test
+    namespace: openshift-migration
+```
+
+- spec.analyzeImageCount: A boolean flag to enable analysis of image count. This is a required field.
+
+- spec.analyzeK8SResources: A boolean flag to enable analysis of k8s resources. This is a required field.
+
+- spec.analyzePVCapacity: A boolean flag to enable analysis of persistent volume capacity. This is a required field.
+
+- spec.listImages: A boolean flag to enable used in analysis of image count.
+
+- spec.listImagesLimit: An integer representing limit on image counts.
+
+- spec.migPlanRef: ObjectReference contains enough information to let you inspect or modify the referred object. This is a required field.
+
+## Mig Cluster
+
+Local Mig cluster: 
+```
+ 
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migcluster-local
+  namespace: openshift-migration
+spec:
+  # [!] Change isHostCluster to 'false' if you want to use a clusterRef and serviceAccountSecretRef
+  #     instead of using the mig-controller detected kubeconfig. Refer to mig-cluster-aws.yaml for an example.
+  isHostCluster: true
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+  ```
+
+  Remote Mig cluster:
+
+  ```
+  apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migcluster-remote
+  namespace: openshift-config
+spec:
+  # [!] Change isHostCluster to 'true' if you'll be running the controller on this cluster.
+  #     Setting 'isHostCluster' to true will bypass using the clusterRef and serviceAccountSecretRef below.
+  isHostCluster: false
+
+  url: "https://my-remote-cluster-ip.nip.io:8443"
+
+  serviceAccountSecretRef:
+    name: sa-token-remote
+    namespace: openshift-config
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+  ```
+- spec.azureResourceGroup: For azure clusters -- it's the resource group that in-cluster volumes use.
+
+- spec.caBundle: If the migcluster needs SSL verification for connections a user can supply a custom CA bundle. This field is required only when spec.Insecure is set false.
+  
+- spec.exposedRegistryPath: Stores the path of registry route when using direct migration.
+
+- spec.insecure: If set false, user will need to provide CA bundle for TLS connection to the remote cluster.
+
+- spec.isHostCluster: Specifies if the cluster is host (where the controller is installed) or not. This is a required field.
+
+- spec.refresh: If set True, forces the controller to run a full suite of validations on migcluster.
+
+- spec.restartRestic: An override setting to tell the controller that the source cluster restic needs to be restarted after stage pod creation.
```

Comment:
@eriknelson @shubham-pampattiwar 
If `insecure` is `true`, does that mean the CA certificates are not checked?
Can `refresh` have to be set on either the source or target cluster? Does this have effects that the user needs to know, like decreased performance?
Does `restartRestic` need to be set on the source or target/host cluster?

Reply:
> spec.insecure: If insecure is true, does that mean the CA certificates are not checked?

If spec.insecure is `true` CA certificates are not required, and not checked as well.

> spec.refresh: Can refresh have to be set on either the source or target cluster? Does this have effects that the user needs to know, like decreased performance?

spec.refresh: can be set `true` for both to run cluster validation. It does not affect the performance, neither this is required during migration.

Reply:
@JaydipGabani 

What is the "stage pod"?

Reply:
Stage pods are dummy pods. These pods mount PVC that are included in migration while pods and other resources are being migrated to make sure nothing breaks.

Reply:
> spec.restartRestic: Does restartRestic need to be set on the source cluster or the target/host cluster? What is the stage pod?

RestartRestic is only relevant for clusters used as source clusters. It has no bearing on the host cluster (unless the source is host), or on the dest/target cluster. It's basically a legacy option for old OCP versions. It's almost never needed since we default to restarting restic for 3.7/2.9, and not elsewhere, but if user have a cluster that needs it that doesn't match this, user can set it.

Reply:
> Stage pods are dummy pods. These pods mount PVC that are included in migration while pods and other resources are being migrated to make sure nothing breaks.

@JaydipGabani  I think the stage pods might be called conversion pods in the documentation. See #8 in the [virt migration workflow diagram](https://konveyor.io/forklift-documentation/documentation/doc-Migration_Toolkit_for_Virtualization/master/index.html#virt-migration-workflows_forklift). Can you confirm?

Reply:
> t's basically a legacy option for old OCP versions. It's almost never needed since we default to restarting restic for 3.7/2.9,

@JaydipGabani  AFAIK, MTC is not supported for OCP versions earlier than 3.9. Is there any reason why it should be documented in downstream docs?

Reply:
MTC is supported back to 3.7. You're linking to the Forklift project, which is MTV (different project). MTC calls these pods stage pods.

Reply:
@eriknelson 
Sorry, got confused with CNV. Yes, different terminology.